skada/metrics.py

```diff
@@ -713,24 +722,30 @@ def _score(self, estimator, X, y=None, sample_domain=None, **params):
         same_idx = (labels_a == labels_b).nonzero()[0]
         diff_idx = (labels_a != labels_b).nonzero()[0]
 
-        # Mixup with images and hard pseudo labels
+        # Mixup with X_target and hard pseudo labels
         mix_inputs = self.alpha * X_target + (1 - self.alpha) * X_target[rand_idx]
-        mix_labels = self.alpha * labels_a + (1 - self.alpha) * labels_b
-
-        # Obtain predictions for the mixed samples
-        mix_pred = estimator.predict(
-            mix_inputs, sample_domain=np.full(mix_inputs.shape[0], -1)
+        lb = LabelBinarizer(neg_label=False, pos_label=True).fit(y[source_idx])
```

Comment:
Not sure that LabelBinarizer will work everytime, why don't you use OneHotEncoder?

Reply:
One hot encoder is made for X not y. But indeed we should change the test for multiple classes.